zulipterminal/ui_tools/boxes.py

```diff
@@ -103,6 +108,19 @@ def stream_box_view(self, stream_id: int, caption: str='', title: str='',
         ]
         self.contents = write_box
 
+    def _stream_box_autocomplete(self, text: str, state: Optional[int]
+                                 ) -> Optional[str]:
+        streams_list = self.view.pinned_streams + self.view.unpinned_streams
+        streams = [stream[0]
+                   for stream in streams_list]
+
+        stream_data = list(zip(streams, streams))
+        matched_streams = match_stream(stream_data, text,
+                                       self.view.pinned_streams)
+
+        return self._process_typeaheads(matched_streams[0], state,
+                                        matched_streams[1])
```

Comment:
`matched_stream[0]` and `matched_stream[1]` have the same data, but I've used both of them for the sake of consistency.

Reply:
Added